Applications/FileIO/SHPInterface.cpp

```diff
@@ -176,17 +178,10 @@ void SHPInterface::readPolygons(const SHPHandle &hSHP, int numberOfElements, con
     auto const polylines = _geoObjects.getPolylineVec(listName);
     auto sfc_vec = std::make_unique<std::vector<GeoLib::Surface*>>();
 
-    for (auto const* polyline : *polylines) {
-        GeoLib::Surface* sfc(GeoLib::Surface::createSurface(*polyline));
-        if (sfc)
-            sfc_vec->push_back(sfc);
-        else {
-            WARN("SHPInterface::readPolygons(): Could not triangulate polygon.")
-        }
+    for (auto const* polyline : *polylines)
+    {
+        FileIO::createSurface(*polyline, _geoObjects, listName);
```

Comment:
the createSurface could fail in multiple ways. I'd keep the testing; maybe return a flag from createSurface...